src/backend/distributed/commands/non_main_db_distribute_object_ops.c

```diff
@@ -335,6 +363,50 @@ DropDbStmtCheckSupportedObjectType(Node *node)
 }
 
 
+static bool
+AlterDbStmtCheckSupportedObjectType(Node *node)
+{
+	/*
+	 * We don't try to send the query to the main database if the ALTER
+	 * DATABASE command is for the main database itself, this is a very
+	 * rare case but it's exercised by our test suite.
+	 */
+	AlterDatabaseStmt *stmt = castNode(AlterDatabaseStmt, node);
+	if (!IsSetTablespaceStatement(stmt))
+	{
+		return true;
+	}
+	else
+	{
+		return IsSetTablespaceStatement(stmt) && strcmp(stmt->dbname, MainDb) != 0;
+	}
+}
+
+
+static bool
+AlterDbCanNotBeExecutedInTransaction(Node *node)
```

Comment:
let's define this after CanNotBeExecutedInTransaction_False function, i.e., at the end of this file